NextcloudTalk/SettingsTableViewController.swift

```diff
@@ -0,0 +1,663 @@
+        }
+        return 0
+    }
+
+    func getIndexPathForConfigurationOption(option: ConfigurationSectionOption) -> IndexPath {
+        let section: Int = getSectionForSettingsSection(section: SettingsSection.kSettingsSectionConfiguration)
+        var optionIndexPath: IndexPath = IndexPath(row: 0, section: section)
+
+        let optionRow = getConfigurationSectionOptions().firstIndex(of: option.rawValue)
+
+        if NSNotFound != optionRow {
+            optionIndexPath = IndexPath(row: optionRow!, section: section)
+        }
+        return optionIndexPath
+    }
+
+    // MARK: User Profile
+
+    func refreshUserProfile() {
+        NCSettingsController.sharedInstance().getUserProfile { _ in
+            self.tableView.reloadData()
+        }
+        self.getActiveUserStatus()
+    }
+
+    func getActiveUserStatus() {
+        let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+        NCAPIController.sharedInstance().getUserStatus(for: activeAccount) { userStatus, error in
+            if error == nil && userStatus != nil {
+                self.activeUserStatus = NCUserStatus(dictionary: userStatus!)
+                self.tableView.reloadData()
+            }
+        }
+    }
+
+    // MARK: Notifications
+
+    @objc func appStateHasChanged(notification: NSNotification) {
+        let appState = notification.userInfo!["appState"]
+        if let appState = appState as? AppState {
+            self.adaptInterfaceForAppState(appState: appState)
+        }
+    }
+
+    @objc func contactsHaveBeenUpdated(notification: NSNotification) {
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+        }
+    }
+
+    @objc func contactsAccessHasBeenUpdated(notification: NSNotification) {
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+        }
+    }
+
+    // MARK: User Interface
+
+    func adaptInterfaceForAppState(appState: AppState) {
+        switch appState {
+        case kAppStateReady:
+            refreshUserProfile()
+        default:
+            break
+        }
+    }
+
+    // MARK: Profile actions
+
+    func userProfilePressed() {
+        let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+        let userProfileVC = UserProfileViewController(account: activeAccount)
+        self.navigationController?.pushViewController(userProfileVC, animated: true)
+    }
+
+    // MARK: User Status
+
+    func presetnUserStatusOptions() {
+        let viewController = UserStatusTableViewController(userStatus: activeUserStatus!)
+        self.navigationController?.pushViewController(viewController, animated: true)
+    }
+
+    // MARK: User phone number
+
+    func checkUserPhoneNumber() {
+        NCSettingsController.sharedInstance().getUserProfile { _ in
+            let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+            if activeAccount.phone.isEmpty {
+                self.presentSetPhoneNumberDialog()
```

Comment:
Was not triggered the first time I changed contactSync value to enabled. Please check.

Reply:
I made a change